File: HelloMetal-OC/HelloMetal-OC/ViewController.m
//
//  ViewController.m
//  HelloMetal-OC
//
//  Created by iann on 2019/7/23.
//  Copyright © 2019 luciglobal. All rights reserved.
//

#import "ViewController.h"
#import "Renderer.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    if (![self.view isKindOfClass:[MTKView class]])
    {
        printf("View attached to ViewController is not an MTKView!");
        return;
    }
    
    mtkView = (MTKView *)self.view;
    id<MTLDevice> device = MTLCreateSystemDefaultDevice();
    NSString *name = [device name];
    NSString *name1 = [mtkView.device name];
    mtkView.device = device;
   
    
    Renderer *renderer = [[Renderer alloc] init: mtkView];
    [renderer mtkView:mtkView drawableSizeWillChange:mtkView.bounds.size];
    mtkView.delegate = renderer;
    [renderer drawInMTKView:mtkView];
}


//- (void)setRepresentedObject:(id)representedObject {
//    [super setRepresentedObject:representedObject];
//
//    // Update the view, if already loaded.
//}


@end

File: HelloMetal-Swift/HelloMetal-Swift/ViewController.swift
//
//  ViewController.swift
//  HelloMetal-Swift
//
//  Created by iann on 2019/7/22.
//  Copyright © 2019 luciglobal. All rights reserved.
//

import Cocoa
import Metal
import MetalKit

class ViewController: NSViewController {
    
    var mtkView : MTKView!
    var renderer: Renderer!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        guard let mtkViewTemp = self.view as? MTKView else {
            print("View attached to ViewController is not an MTKView!")
            return
        }
        mtkView = mtkViewTemp
        
        guard let defaultDevice = MTLCreateSystemDefaultDevice() else {
            print("Metal is not supported on this device.")
            return
        }
        
        print("My GPU is: \(defaultDevice)")
        mtkView.device = defaultDevice
        
        guard let tempRederer = Renderer(mtkView: mtkView) else {
            print("Renderer failed to initialize")
            return
        }
        
        renderer = tempRederer
        mtkView.delegate = renderer
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }


}

